x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/esnative/NativeUsersStore.java

```diff
@@ -283,6 +286,25 @@ private void createReservedUser(String username, char[] passwordHash, RefreshPol
         });
     }
 
+    /**
+     * Asynchronous method to create a security index if necessary and a reserved user if one doesn't exist
+     */
+    public void createReservedUserAndGetUserInfo(String username, char[] passwordHash, RefreshPolicy refresh,
+                                                ActionListener<ReservedUserInfo> listener) {
+        securityIndex.prepareIndexIfNeededThenExecute((e) -> { listener.onFailure(new ElasticsearchStatusException(e.getMessage(),
+                INTERNAL_SERVER_ERROR, e.getCause())); },
+            () -> { executeAsyncWithOrigin(client.threadPool().getThreadContext(), SECURITY_ORIGIN,
+                client.prepareIndex(SECURITY_MAIN_ALIAS).setOpType(DocWriteRequest.OpType.CREATE)
+                    .setId(getIdForUser(RESERVED_USER_TYPE, username))
+                    .setSource(Fields.PASSWORD.getPreferredName(), String.valueOf(passwordHash),
+                        Fields.ENABLED.getPreferredName(),
+                        true, Fields.TYPE.getPreferredName(), RESERVED_USER_TYPE)
+                    .setRefreshPolicy(refresh).request(),
+                listener.<IndexResponse>delegateFailure((l, indexResponse) -> getReservedUserInfo(username, l)),
```

Comment:
The get after a successful create is unnecessary.

Reply:
See ^^